src/vm/contexts.rs

```diff
@@ -224,11 +354,19 @@ impl <'a> GlobalContext <'a> {
     
     pub fn new(database: ContractDatabase<'a>) -> GlobalContext<'a> {
         GlobalContext {
+            parent_map: None,
             database: database,
-            read_only: false
+            read_only: false,
+            asset_map: AssetMap::new()
         }
     }
 
+    pub fn log_asset_transfer(&mut self, sender: &PrincipalData, contract_name: &str, asset_name: &str, transfered: i128) -> Result<()> {
```

Comment:
Why is this method called `log`?  Is it part of an event log or something?  Just curious about how to interpret the name.

Reply:
It's not actually performing an asset transfer, merely logging the fact that one has occurred.